### ISSUE TITLE
Incorrect computation of register space causes panic

### ISSUE THREAD

**Issue by h1048576** [cel-expr/cel-go#123]
The following program results in a panic when attempting to update the state during 
one of the comprehensions.

```
package main

import (
	"fmt"

	"github.com/google/cel-go/checker"
	"github.com/google/cel-go/checker/decls"
	"github.com/google/cel-go/common"
	"github.com/google/cel-go/common/packages"
	"github.com/google/cel-go/common/types"
	"github.com/google/cel-go/interpreter"
	"github.com/google/cel-go/parser"
)

func main() {
	ss := `code == "111" && ["a", "b"].all(x, x in tags) 
                  || code == "222" && ["a", "b"].all(x, x in tags)`
	//src := common.NewTextSource("a || b && c.exists(x, x > 2)")
	src := common.NewTextSource(ss)
	expr, errors := parser.Parse(src)
	if len(errors.GetErrors()) != 0 {
		fmt.Println(errors.ToDisplayString())
		return
	}

	typeProvider := types.NewProvider()
	env := checker.NewStandardEnv(packages.DefaultPackage, typeProvider)
	env.Add(
		decls.NewIdent("code", decls.String, nil),
		decls.NewIdent("tags", decls.NewListType(decls.String), nil),
	)
	c, errors := checker.Check(expr, src, env)
	if len(errors.GetErrors()) != 0 {
		fmt.Println(errors.ToDisplayString())
		return
	}

	i := interpreter.NewStandardInterpreter(packages.DefaultPackage, typeProvider)
	eval := i.NewInterpretable(interpreter.NewCheckedProgram(c))
	result, state := eval.Eval(
		interpreter.NewActivation(
			map[string]interface{}{
				"code": "222",
				"tags": []string{"a", "b"}}))

	fmt.Println(state)
	fmt.Println(result)
}
```

**Comment by h1048576** [comment]
package main

import (
	"fmt"

	"github.com/google/cel-go/checker"
	"github.com/google/cel-go/checker/decls"
	"github.com/google/cel-go/common"
	"github.com/google/cel-go/common/packages"
	"github.com/google/cel-go/common/types"
	"github.com/google/cel-go/interpreter"
	"github.com/google/cel-go/parser"
)

func main() {

	ss := `
code == "111" && ["a", "b"].all(x, x in tags)
|| code == "222" && ["a", "b"].all(x, x in tags)
`
	//src := common.NewTextSource("a || b && c.exists(x, x > 2)")
	src := common.NewTextSource(ss)
	expr, errors := parser.Parse(src)
	if len(errors.GetErrors()) != 0 {
		fmt.Println(errors.ToDisplayString())
		return
	}

	typeProvider := types.NewProvider()
	env := checker.NewStandardEnv(packages.DefaultPackage, typeProvider)
	env.Add(
		decls.NewIdent("code", decls.String, nil),
		decls.NewIdent("tags", decls.NewListType(decls.String), nil),
	)
	c, errors := checker.Check(expr, src, env)
	if len(errors.GetErrors()) != 0 {
		fmt.Println(errors.ToDisplayString())
		return
	}

	i := interpreter.NewStandardInterpreter(packages.DefaultPackage, typeProvider)
	eval := i.NewInterpretable(interpreter.NewCheckedProgram(c))
	result, state := eval.Eval(
		interpreter.NewActivation(
			map[string]interface{}{
				"code": "222",
				"tags": []string{"a", "b"}}))

	fmt.Println(state)
	fmt.Println(result)
}

**Comment by h1048576** [comment]
if ss like this, it is ok
ss := `
code == "111"
|| code == "222" && ["a", "b"].all(x, x in tags)
`

**Comment by TristonianJones** [comment]
I believe you have encountered a bug addressed in the pull request for issue #119. I'll try to reproduce your issue before submitting the PR.

**Comment by TristonianJones** [comment]
Looks to be a different issue where the internal register space doesn't account for internal variables needed to evaluate the comprehensions. It happens that there was one extra register allocated, but with two `all` comprehensions, the internal variable count exceeds the allocated limit. 

Thanks for the report! I'll get to this later today.
